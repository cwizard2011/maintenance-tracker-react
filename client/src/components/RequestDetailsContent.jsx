import React from 'react';
import PropTypes from 'prop-types';

const RequestDetailsContent = (props) => {
  const {
    title, requestId, labelClass, labelName, status, details, created, updated
  } = props;
  return (
    <div className="request-details">
      <h3>Request Details</h3>
      <p>
      Title:
        {' '}
        <span>
          {title}
        </span>
      </p>
      <p>
      Request Id:
        {' '}
        <span>
          {requestId}
        </span>
      </p>
      <p>
      Status:
        {' '}
        <span className={labelClass}>
          <ion-icon name={labelName} />
          {status}
        </span>
      </p>
      <p>
      Details:
        {' '}
        <span>
          {details}
        </span>
      </p>
      <p>
      Created on:
        {' '}
        <span>
          {' '}
          {created}
          {' '}
        </span>
      </p>
      <p>
      Updated on:
        {' '}
        <span>
          {' '}
          {updated}
          {' '}
        </span>
      </p>
    </div>
  );
};
RequestDetailsContent.propTypes = {
  title: PropTypes.string,
  requestId: PropTypes.string,
  labelClass: PropTypes.string,
  labelName: PropTypes.string,
  status: PropTypes.string,
  details: PropTypes.string,
  created: PropTypes.string,
  updated: PropTypes.string
};
export default RequestDetailsContent;
