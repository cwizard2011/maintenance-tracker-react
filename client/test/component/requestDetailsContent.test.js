import React from 'react';
import { shallow } from 'enzyme';
import RequestDetailsContent from '../../src/components/RequestDetailsContent';

describe('Request detail', () => {
  it('render request detail correctly', () => {
    const wrapper = shallow(<RequestDetailsContent />);
    expect(wrapper).toMatchSnapshot();
  });
});
