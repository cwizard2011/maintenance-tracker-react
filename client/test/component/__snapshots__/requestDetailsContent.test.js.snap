// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request detail render request detail correctly 1`] = `
<div
  className="request-details"
>
  <h3>
    Request Details
  </h3>
  <p>
    Title:
     
    <span />
  </p>
  <p>
    Request Id:
     
    <span />
  </p>
  <p>
    Status:
     
    <span>
      <ion-icon />
    </span>
  </p>
  <p>
    Details:
     
    <span />
  </p>
  <p>
    Created on:
     
    <span>
       
       
    </span>
  </p>
  <p>
    Updated on:
     
    <span>
       
       
    </span>
  </p>
</div>
`;
